src/14/HW/TodoList.ts

```diff
@@ -0,0 +1,66 @@
+import {Note} from "./Note";
+
+class TodoList {
+    private notes: Note[] = [];
+
+    public addNote(note: Note) {
+        this.notes.push(note);
+    }
+
+    public removeNote(id: number): boolean {
+        const index = this.notes.findIndex(note => note.id === id);
+        if (index !== -1) {
+            this.notes.splice(index, 1);
+            return true;
+        }
+        return false;
+    }
+
+    public editNote(id: number, newContent: string): boolean {
+        const note = this.notes.find(note => note.id === id);
+        if (note) {
+            note.updateContent(newContent);
+            return true;
+        }
+        return false;
+    }
+
+    public getNoteById(id: number): Note | undefined {
+        return this.notes.find(note => note.id === id);
+    }
+
+    public getAllNotes(): Note[] {
+        return this.notes;
+    }
+
+    public markNoteAsCompleted(id: number): boolean {
+        const note = this.notes.find(note => note.id === id);
+        if (note) {
+            note.markAsCompleted();
+            return true;
+        }
+        return false;
+    }
+
+    public getTotalNotes(): number {
+        return this.notes.length;
+    }
+
+    public getPendingNotesCount(): number {
+        return this.notes.filter(note => note.status === 'active').length;
+    }
+
+    public searchNotes(keyword: string): Note[] {
+        return this.notes.filter(note =>
+            note.title.includes(keyword) || note.content.includes(keyword)
+        );
+    }
+
+    public sortNotes(by: 'status' | 'createdAt') {
```

Comment:
The same. The sorting feature needs to be implemented separately from the TodoList entity. This means that it should extend the TodoList class. Please, double-check the requirements.